graphql_compiler/schema_transformation/rename_schema.py

```diff
@@ -277,4 +293,38 @@ def enter_FieldDefinition(self, node, *args):
         if self.in_query_type:
             field_name = node.name.value
             new_field_name = self.renamings.get(field_name, field_name)  # Default use original
-            node.name.value = new_field_name
+            if new_field_name == field_name:
+                return None
+            else:  # Make copy of node with the changed name, return the copy
+                field_node_with_new_name = _get_copy_of_node_with_new_name(node, new_field_name)
+                return field_node_with_new_name
+
+
+def _get_copy_of_node_with_new_name(node, new_name):
+    """Return a node with new_name as its name and otherwise identical to the input node.
+
+    Args:
+        node: type Node, with a .name attribute. Not modified by this function
+        new_name: str, name to give to the output node
+
+    Returns:
+        Node, with new_name as its name and otherwise identical to the input node
+    """
+    node_type = type(node).__name__
+    allowed_types = frozenset((
+        'EnumTypeDefinition',
+        'FieldDefinition',
+        'InterfaceTypeDefinition',
+        'NamedType',
+        'ObjectTypeDefinition',
+        'UnionTypeDefinition',
+    ))
+    if node_type not in allowed_types:
+        raise AssertionError(
+            u'Input node {} of type {} is not allowed, only {} are allowed.'.format(
+                node, type(node).__name__, allowed_types
```

Comment:
nit: I think `type(node).__name__` is already in scope as `node_type`, might be a good idea to reuse it.